### PR TITLE
Polish GHA config

### PR DESCRIPTION
**Pull request by Goooler**
- Use the latest stable OS on CI.
- Use the latest actions/setup-node.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -22,11 +22,11 @@ jobs:
           GH_TOKEN=${{ secrets.GITHUB_TOKEN }} npm run publish-linux-app
 
   macos:
-    runs-on: macos-10.15
+    runs-on: macos-latest
```
Comment (aaronliu0130) [inline .github/workflows/main.yml:25]: This leads to issues such as #98.

Comment (Goooler) [reply .github/workflows/main.yml]: 303aabd0d0668ea8f6ba1a8afee121a78f603d76

**Comment by Goooler** [comment]
`ubuntu-latest` is the latest LTS, see https://github.blog/changelog/2022-11-09-github-actions-ubuntu-latest-workflows-will-use-ubuntu-22-04/

**Comment by Goooler** [comment]
`ubuntu-latest` is the latest LTS, see https://github.blog/changelog/2022-11-09-github-actions-ubuntu-latest-workflows-will-use-ubuntu-22-04

**Comment by NayamAmarshe** [comment]
Okay, wait. Now I remember. We cannot use Ubuntu latest because using Ubuntu 18.04 ensures that the AppImage can work with 18.04 and up.

**Comment by aaronliu0130** [comment]
Are programs/appimages compiled on later versions unable to be run on older versions?

**Comment by NayamAmarshe** [comment]
> Are programs/appimages compiled on later versions unable to be run on older versions?

Yeah, I read that somewhere.

**Comment by aaronliu0130** [comment]
How about setup-node? Note that we'd need a squash at this point if only setup-node is to be kept.

**Comment by aaronliu0130** [comment]
Also, do we still need to support 18.04? That would reach its EOL in 17 days.

**Comment by NayamAmarshe** [comment]
> Also, do we still need to support 18.04? That would reach its EOL in 17 days, and the next release would probably only occur after that.

I think it won't make a difference for a while. There might be a few people using 18.04 so I think we should keep it.

**Pull request closed without merging**